﻿#READMEloading packages```library(dplyr)library(plyr)```Reading data from git hub repository````fang <- read.delim("https://raw.githubusercontent.com/musimwa/BCB546X_FINAL-UNIX-ASSIGNMENT/master/dataset/fang_et_al_genotypes.txt", header = T)## snp <- read.delim("https://raw.githubusercontent.com/musimwa/BCB546X_FINAL-UNIX-ASSIGNMENT/master/dataset/snp_position.txt", header = T)````## Data inspection fang_ et _al```{r}head(fang, n=10)str(fang)dim(fang)class(fang)typeof(fang)levels(fang$Group)colnames(fang)print(object.size(fang), units="Mb")```## Data inspection snp_position```{r}head(snp, n=1)str(snp)dim(snp)class(snp)typeof(snp)levels(snp$Chromosome)colnames(snp)print(object.size(snp), units="Mb")```## Data processingGrouping maize```{r}maize_grps <- c("ZMMIL", "ZMMLR", "ZMMMR")maize <- fang[fang$Group %in% maize_grps, ] dim(maize)```Grouping teosinte```{r}teosinte_grps <- c("ZMPBA", "ZMPIL", "ZMPJA")teosinte <- fang[fang$Group %in% teosinte_grps, ] dim(maize) ```TransposingThe 1st three rows are not necessary and are removed```{r}maize_transp <-as.data.frame(t(maize[,-c(1:3)])) dim(maize_transp)``````{r}teosinte_transp <- as.data.frame(t(teosinte[,-c(1:3)])) dim(teosinte_transp)```Editing the SNP data file```{r}final_pos <- subset(snp[,c(1,3,4)])dim(final_pos)```Sorting the data file```{r}maize_sort <- maize_transp[order(row.names(maize_transp)),]teosinte_sort <- teosinte_transp[order(row.names(teosinte_transp)),]position_sort <- final_pos[order(final_pos$SNP_ID),]; position_sort```Filtering```{r}chromosome_names <- c(1,2,3,4,5,6,7,8,9,10)position_sort_filter <- subset(position_sort, Chromosome %in% chromosome_names)dim(position_sort_filter)position_sort_filter[1:3, 1:3]```Dataset merger```{r}maize_join <- merge(position_sort_filter, maize_sort, by.x ="SNP_ID", by.y ="row.names") dim(maize_join)teosinte_join <- merge(position_sort_filter, teosinte_sort, by.x ="SNP_ID", by.y ="row.names")  dim(teosinte_join)```Ascending order sorting with question mark (?) or dash (-)```{r}maize_sort_asc <- arrange(maize_join, Position) # sort ascendingly by positionmaize_sort_asc[1:9, 1:9] # check the dataset``````{r}teosinte_sort_asc <- arrange(teosinte_join, Position) teosinte_sort_asc[1:9, 1:9] ```Replacing (?) with (-) using (gtools)```{r}maize_join_dash <-gsub("?", " -", maize_join)teosinte_join_dash <-gsub("?", " -", teosinte_join)```Sorting SNP position in descending order using (-)```{r}maize_sort_desc_dash <- arrange(maize_join, desc(Position))teosinte_sort_desc_dash <- arrange(teosinte_join, desc(Position))```Chromosome extraction in ascending order (?)```{r}for(i in 1:10){  chrom_extract_asc<-maize_sort_asc[maize_sort_asc$Chromosome ==i,]  write.csv(chrom_extract_asc, paste("maize_chr_ascend.csv",i,".csv",sep=""),row.names=F)}``````{r}for(i in 1:10){  chrom_extract_asc<-teosinte_sort_asc[teosinte_sort_asc$Chromosome==i,]  write.csv(chrom_extract_asc, paste("teosinte_chr_ascend.csv",i,".csv",sep=""),row.names=F)}```Chromosone extraction in descending order (-)```{r}for(i in 1:10){  chrom_extract_desc<-maize_sort_desc_dash[maize_sort_desc_dash$Chromosome ==i,]  write.csv(chrom_extract_desc, paste("maize_chr_descend.csv",i,".csv",sep=""),row.names=F)}``````{r}for(i in 1:10){  chrom_extract_desc<-teosinte_sort_desc_dash[teosinte_sort_desc_dash$Chromosome ==i,]  write.csv(chrom_extract_desc, paste("maize_chr_descend",i,".csv",sep=""),row.names=F)}```#Part 2Loading packages```{r}library(ggplot2)library(reshape2)```Transposing original data```{r}fang_transposed<-as.data.frame(t(fang))dim(fang_transposed)```Merge fang_ transposed with snp_position data```{r}data_joined<-merge(snp, fang_transposed, by.x="SNP_ID",by.y="row.names",all=TRUE)data_joined<-data_joined[-c(984:986),]```Visualizations in ggplot2```{r}data_joined$Chromosome<-factor(data_joined$Chromosome, levels = c("1","2","3","4","5","6","7","8","9","10","unknown","multiple"))ggplot(data_joined)+ geom_bar(aes(data_joined$Chromosome))+xlab("Chromosome") +ylab("total number of SNPs")```Group with most SNPs```{r}ggplot(fang)+geom_bar(aes(fang$Group))+xlab("SNP group")+ylab("total SNPs")```Missing data and amount of heterozygosity```{r}geno <- colnames(fang)[-c(1:3)]geno_melted<-melt(fang,measure.vars = geno)geno_meltedgeno_melted[geno_melted =="?/?"]=NA```creating column for homo & heterozygotes using true/false $ NA for missing data```{r}geno_melted$Homozygotes_heterozygotes<-(geno_melted$value=="A/A"|geno_melted$value=="C/C"|geno_melted$value=="G/G"|geno_melted$value=="T/T")geno_melted_sorted<-arrange(geno_melted,Sample_ID,Group) #sorting by sample ID & group```plotting the homozygotes, heterozygotes and NAs```{r}read_indv<-ddply(geno_melted_sorted,c("Sample_ID"),summarise,total_homozygous=sum(Homozygotes_heterozygotes,na.rm=TRUE),total_heterozygous=sum(!Homozygotes_heterozygotes,na.rm = TRUE), total_NA=sum(is.na(Homozygotes_heterozygotes)))read_indv_melted<-melt(read_indv,measure.vars = c("total_homozygous","total_heterozygous","total_NA"))ggplot(read_indv_melted,aes(x=Sample_ID,y=value,fill=variable))+geom_bar(stat="identity",position = "stack")#plot homozygotes and heterozygotes per sample_ID```Homozygotes & heterozygotes per group```{r}group<-ddply(geno_melted_sorted,c("Group"),summarise,total_homozygous=sum(Homozygotes_heterozygotes,na.rm=TRUE),total_heterozygous=sum(!Homozygotes_heterozygotes,na.rm = TRUE), total_NA=sum(is.na(Homozygotes_heterozygotes)))grp_melted<-melt(group,measure.vars = c("total_homozygous","total_heterozygous","total_NA"))ggplot(grp_melted,aes(x=Group,y=value,fill=variable))+geom_bar(stat="identity",position = "stack")# plot homozygotes and heterozygotes per Group```Own visualizationCandidate.random vs gene```{r}data_joined$candidate.random<-factor(data_joined$candidate.random, levels = c("candidate","random"))ggplot(data_joined)+ geom_bar(aes(data_joined$candidate.random,fill = candidate.random))+xlab("candidate.random") +ylab("Total Number of SNPs") 